gh-110527: Improve `PySet_Clear` docs (#110528)
python/cpython: Doc/c-api/set.rst

.. c:function:: int PySet_Clear(PyObject *set)

   Empty an existing set of all elements. Return ``0`` on

>    success. If *set* is not an instance of :class:`set` or its subtype

Reviewer: ```suggestion
   success. If *set* is not an instance of :class:`set` or a :class:`set` subclass,
```

>    instance of :class:`set` or its subtype.

Reviewer: This doc is quite generic and doesn't explain that it cannot fail if you pass a set type or a subtype of the set type.

It only fails the argument type is not set.

Author: Done, should be better now!
